skyreader/utils/handle_map_data.py

```diff
@@ -25,6 +27,8 @@ def extract_map(
             sigma of the gaussian to convolute the map with in deg.
         - remove_min_val: bool = True. Remove minimum value from -llh
           no effect if probability map.
+        - return_uniqs: bool = True. Return the uniqs for each pixel
+            (useful for multiorder maps)
```

Comment:
This is an antipattern. Functions should always return the same type of object (before, this was a 3-tuple). I recommend type-hinting the signature's return type.

A wrapper function that returns a 4-tuple would suffice for your multiorder-map use case.

Reply:
I understand that it is a problem that the function sometimes returns a 4-tuple and sometimes a 5-tuple (before it was a 4-tuple and now it is a 5-tuple). However, I am not sure what do you mean with implementing a wrapper function that returns a 4-tuple. Do you mean a function that runs `extract_map` in some special way so that it returns also the uniq list? Or that afterwards uses the results to obtain also the uniqs? It is not super simple to deduce afterwards the uniqs, it is necessary to intervene at several points inside the cycles. Not sure I get at what level you want me to wrap the functions to make it return the uniqs only in a special case.

Reply:
I propose an additional function called `extract_map_with_uniqs()` which returns a 5-tuple. `extract_map()` always returns a 4-tuple. The exact implementation is up to you--reading the lines below, it looks like `extract_map()` would actually call to `extract_map_with_uniqs()` but ignore the returned "uniq_array" (so my original suggestion was backward).

Or, if you're okay with always making a "uniq_array", change the function to always return a 5-tuple. And update all the existing calls, ignoring the value when not needed. For example: 
```python
grid_value, grid_ra, grid_dec, equatorial_map, _ = extract_map(
    result,
    llh_map,
    angular_error_floor,
    remove_min_val=not llh_map,
)
```

Reply:
Implemented! Nor it always returns a 5 Tuple